pymc4/inference/tensorflow/sampling.py

```diff
@@ -88,34 +89,42 @@ def sample(
 
     """
     logpfn, init = build_logp_function(model, state=state, observed=observed)
-    init = [tf.tile(tf.expand_dims(tens, 0), [num_chains] + [1] * tens.ndim) for tens in init]
+    init = [
+        tf.tile(tf.expand_dims(tens, 0), [num_chains] + [1] * tens.ndim)
+        for tens in init
+    ]
+
+    def parallel_logpfn(*state):
```

Comment:
Could we isolate this function to Model as a property? it is otherwise difficult to test.